samples/TestApp.PCL/TokenBroker.cs

```diff
@@ -93,67 +95,84 @@ UserIdentifier GetUserIdentifier()
         {
             try
             {
-                context = new AuthenticationContext(Sts.Authority, true);
+                AuthenticationContext context = new AuthenticationContext(Sts.Authority, true);
                 var result = await context.AcquireTokenSilentAsync(Sts.ValidResource, Sts.ValidClientId, GetUserIdentifier(), parameters);
                 return result.AccessToken;
             }
             catch (Exception ex)
             {
-                string msg = ex.Message + "\n" + ex.StackTrace;
-
-                return msg;
+                return GetErrorMessage(ex);
             }
         }
 
         public async Task<string> GetTokenInteractiveAsync(IPlatformParameters parameters)
         {
             try
             {
-                context = new AuthenticationContext(Sts.Authority, true);
+                AuthenticationContext context = new AuthenticationContext(Sts.Authority, true);
                 var result = await context.AcquireTokenAsync(Sts.ValidResource, Sts.ValidClientId, Sts.ValidNonExistingRedirectUri, parameters, GetUserIdentifier());
                 return result.AccessToken;
             }
             catch (Exception ex)
             {
-                string msg = ex.Message +"\n"+ex.StackTrace;
-                
-                return msg;
+                return GetErrorMessage(ex);
             }
         }
 
         public async Task<string> GetTokenInteractiveWithMsAppAsync(IPlatformParameters parameters)
         {
             try
             {
-                context = new AuthenticationContext(Sts.Authority, true);
+                AuthenticationContext context = new AuthenticationContext(Sts.Authority, true);
                 var result = await context.AcquireTokenAsync(Sts.ValidResource, Sts.ValidClientId, null, parameters, GetUserIdentifier());
 
                 return result.AccessToken;
             }
             catch (Exception ex)
             {
-                return ex.Message;
+                return GetErrorMessage(ex);
             }
         }
 
         public async Task<string> GetTokenWithClientCredentialAsync()
         {
             try
             {
-                context = new AuthenticationContext(Sts.Authority, true);
+                AuthenticationContext context = new AuthenticationContext(Sts.Authority, true);
                 var result = await context.AcquireTokenAsync(Sts.ValidResource, new ClientCredential(Sts.ValidConfidentialClientId, Sts.ValidConfidentialClientSecret));
 
                 return result.AccessToken;
             }
             catch (Exception ex)
             {
-                return ex.Message;
+                return GetErrorMessage(ex);
             }
         }
 
         public void ClearTokenCache()
         {
             TokenCache.DefaultShared.Clear();
         }
+
+        private static string GetErrorMessage(Exception ex)
```

Comment:
Improves the error message returned to simplify debugging. The error message now includes all of the messages from any internal exceptions.